### PR TITLE
NEW Well Index Calculation for segmented wells

### PR DESCRIPTION
**Pull request by einar90**
This is the contents of #3 with some small fixes (mainly in unit tests).

If it's okay for @agchitu , I will close #3 and merge this one.

### CODE REVIEW

File: intersected_cell.cpp [hunk 1]
```diff
@@ -1,78 +1,108 @@
...
-
-   FieldOpt is distributed in the hope that it will be useful,
-   but WITHOUT ANY WARRANTY; without even the implied warranty of
-   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-   GNU General Public License for more details.
-
-   You should have received a copy of the GNU General Public License
-   along with FieldOpt.  If not, see <http://www.gnu.org/licenses/>.
-******************************************************************************/
-
```
Comment (Bellout) [inline intersected_cell.cpp]: Why is the license text removed?

File: main.cpp [hunk 2]
```diff
@@ -30,34 +31,96 @@
...
-    string well_name = vm["well-name"].as<string>();
-    printCompdat(well_blocks, well_name, wellbore_radius);
-  }
-    // Otherwise, print as a CSV table
-  else {
-    printCsv(well_blocks);
-  }
-
-  return 0;
+int main(int argc, const char *argv[])
```
Comment (Bellout) [inline main.cpp]: Run auto-indent on this file.

File: main.hpp [hunk 3]
```diff
@@ -51,162 +56,190 @@
...
+    
+    // If called with --help or -h flag:
+    if (vm.count("help") || !success_arg) 
+    { // Print help if --help present or input file/output dir not present
+        cout << "Usage: ./WellIndexCalc --grid gridpath --heel x1 y1 z1 --toe x2 y2 z2 --radius r [options]" << endl;
+        cout << "options can be --compdat --well-name Name" << endl;
+        cout << "Or" << endl;
+        cout << "Usage: ./WellIndexCalc --grid gridpath --well-filedef filepath" << endl;
+        cout << desc << endl;
+
```
Comment (Bellout) [inline main.hpp]: I think in this loop we should add the debug code above that tells us specifically which parameter that are missing/wrong

File: main.hpp [hunk 4]
```diff
@@ -51,162 +56,190 @@
...
+    }
 }
 
-void printCompdat(vector<IntersectedCell>& well_blocks,
-                  string well_name,
-                  double wellbore_radius) {
-  string head = "COMPDAT\n";
-  string foot = "\n/";
-  vector<string> body;
-  //                      NAME   I   J  K1  K2  OP/SH ST  WI   RADIUS
```
Comment (Bellout) [inline main.hpp]: I think we should re-add this link, one extra line but it enables the following code to be more compact

Comment (Bellout) [inline main.hpp]: -  //                      NAME   I   J  K1  K2  OP/SH ST  WI   RADIUS
 -  string compdat_frmt = "   %s  %d  %d  %d  %d  OPEN  1*  %8s  %8s /";

Comment (einar90) [inline main.hpp]: I disagree. The way it is now is the way string builders like this are usually written.

File: main.hpp (hunk 3 above)
Comment (einar90) [inline main.hpp]: I agree, I'll fix it.

File: intersected_cell.cpp (hunk 1 above)
Comment (einar90) [inline intersected_cell.cpp]: I'll re-add it.

File: tests/test_single_cell_wellindex.cpp [hunk 5]
```diff
@@ -57,7 +59,7 @@ TEST_F(SingleCellWellIndexTest, WellIndexValueWithQVector_test) {
     //double kx = 1.689380;
     //double ky = 1.689380;
     //double kz = 1;
-    //Figure out conversions and stuff?
+    //Figure out conversions and shit?
```
Comment (einar90) [inline tests/test_single_cell_wellindex.cpp]: I agree to this change.

**Pull request merged**